Content.Server/Gatherable/GatherableSystem.cs

```diff
@@ -49,51 +45,49 @@ private void OnInteractUsing(EntityUid uid, GatherableComponent component, Inter
             BreakOnTargetMove = true,
             BreakOnUserMove = true,
             MovementThreshold = 0.25f,
+            DuplicateCondition = DuplicateConditions.SameTarget,
         };
 
         _doAfterSystem.TryStartDoAfter(doAfter);
     }
 
-    private void OnDoAfter(EntityUid uid, GatherableComponent component, GatherableDoAfterEvent args)
+    private void OnInteractHand(EntityUid uid, GatherableComponent component, InteractHandEvent args)
     {
-        if(!TryComp<GatheringToolComponent>(args.Args.Used, out var tool))
+        if (component.ToolWhitelist?.Tags?.Contains("Hand") == false)
             return;
 
-        tool.GatheringEntities.Remove(uid);
+        var doAfter = new DoAfterArgs(args.User, TimeSpan.FromSeconds(component.HarvestTimeByHand), new GatherableDoAfterEvent(), uid, target: uid)
+        {
+            BreakOnDamage = true,
+            BreakOnTargetMove = true,
+            BreakOnUserMove = true,
+            MovementThreshold = 0.25f,
+            DuplicateCondition = DuplicateConditions.SameTarget,
+        };
+
+        _doAfterSystem.TryStartDoAfter(doAfter);
+    }
+
+    private void OnDoAfter(EntityUid uid, GatherableComponent component, GatherableDoAfterEvent args)
+    {
+        if (TryComp<GatheringToolComponent>(args.Args.Used, out var tool))
+            tool.GatheringEntities.Remove(uid);
+
         if (args.Handled || args.Cancelled)
             return;
 
-        Gather(uid, args.Args.Used, component, tool.GatheringSound);
+        Gather(uid, args.Args.Used, component, tool?.GatheringSound);
         args.Handled = true;
```

Comment:
Don't use nullable here, just return if it's not found.

Reply:
In this case, won't it always return if hands are being used to gather?